slatedb/src/garbage_collector/compacted_gc.rs

```diff
@@ -61,13 +65,45 @@ impl CompactedGcTask {
         }
         Ok(active_ssts)
     }
+
+    /// Returns a `DateTime<Utc>` barrier based on the compactor's oldest running compaction start.
+    ///
+    /// When compactions are active (compactor/running_compactions > 0), we read
+    /// compactor/compaction_low_watermark_ts and convert it into a
+    /// `DateTime<Utc>` value. GC should not delete any compacted SST whose ULID timestamp
+    /// is greater than or equal to this barrier time.
+    ///
+    /// This is a process-local coordination mechanism that only works when the compactor
+    /// and garbage collector run in the same process and share the same StatRegistry. It's
+    /// a hack until we have proper compactor persistence (so GC can retrieve the compactor
+    /// state from the object store). See #604 for details.
+    fn compaction_low_watermark_dt(&self) -> Option<DateTime<Utc>> {
+        let running = self
+            .stat_registry
+            .lookup(RUNNING_COMPACTIONS)
+            .map(|s| s.get())
+            .unwrap_or(0);
+        if running <= 0 {
+            return None;
+        }
```

Comment:
curious why we're tracking `COMPACTION_LOW_WATERMARK_TS` separately from `RUNNING_COMPACTIONS` - this could easily be a method that computes the min_ts every time instead of persisting it (which I'm always anxious about because it means we have to have logic like this where if we forget to check `jobs.is_empty()` we might never clean things up